openwisp_utils/admin_theme/menu.py

```diff
@@ -0,0 +1,154 @@
+from django.apps import registry
+from django.core.exceptions import ImproperlyConfigured
+from django.urls import reverse
+from django.utils.translation import gettext as _
+
+from ..utils import SortedOrderedDict
+
+MENU = SortedOrderedDict()
```

Comment:
A global variable, is there another way?
Perhaps a singleton?

Reply:
Global variables are evil. We should try to avoid them when possible and only use them when there is no other way.
I think we can continue for now, this is a lower priority issue.
@nemesisdesign What do you think? Should we log an issue for creating a singleton class fir managing this?

P.S: Global variables are evil because their scope is too big and we can't limit it's access/editing, a singleton can give us some control over it's access! :smile:

Reply:
Thanks @atb00ker 
Lets do it in a separate PR. For now I am updating this PR so that we can proceed further.

Reply:
how are we currently doing it for the current menu? Or for the dashboard with the pie charts?

@atb00ker I would not look into this as something that is evil or not, if it's useful I would use it, if it's harmful or has downsides that affect us then we should avoid it. A data structure which is accessible from code is not something new in python, which does not have any real private methods.

If we want to improve it, we can create a singletone and try to hide the datastructure in the singleton, providing methods to manipulate it, just let's make sure it's consistent with the dashboard code and not end up with having it done in way in one place and another way in another place.

Reply:
Agreed, Singleton implementation is implementing at lot of places; we can come back to it in the future if we have time. (Nice to have task)